### PR TITLE
docs: recommend using a locale object instead of using i18next FE-3983

### PR DESCRIPTION
**Pull request by jamime**
Update the RFC to recommend shipping a local object instead of using `i18next`.

---
This PR does not follow the typical PR process. Details of the RFC process can be found [here](https://github.com/Sage/carbon/blob/master/rfcs/README.md).

Please use 👍  if you approve.
If you don't approve please comment below.

### CODE REVIEW

**Review by mmaciorowski-sage — APPROVED** [review]

**Review by mmaciorowski-sage — DISMISSED** [review]

**Review by thehiggster — DISMISSED** [review]

**Review by sjohnson-sage — APPROVED** [review]

File: rfcs/text/i18n.md
Comment (chrisbarber86) [inline rfcs/text/i18n.md:11]: Potentially this could add quite a bit of overhead going forwards. For example if we add the following:

* en-GB
* en-US
* en-CA
* fr-CA
* fr-FR
* de-DE
* es-ES
* es-US

Every time a new component gets added, or an existing one is changed - we will need to enforce this line about `complete locales` - requiring that the translations are updated for all those locales (which isn't an exhaustive list). These would also require review - which the core team might not be able to do without outside help

For me it would make sense for Carbon to default always to english, and allow consumers an easy way to provide translations for their own needs in their application (e.g. using I18next or I18n.js)

File: rfcs/text/i18n.md
Comment (chrisbarber86) [inline rfcs/text/i18n.md:31]: I would suggest providing a locale is the main mechanism and providing props is seen more for edge cases where the content is quite different for a particular scenario. Overriding translations via props would work, but wouldn't scale particularly well.

File: rfcs/text/i18n.md
Comment (dawidzarzycki) [inline rfcs/text/i18n.md:119]: is these two styles of comments are on purpose?

**Review by chrisbarber86 — CHANGES_REQUESTED** [review]

File: rfcs/text/i18n.md
Comment (DlgSHi) [inline rfcs/text/i18n.md:62]:
```suggestion
  locale: "en-Gb",
```

**Review by DlgSHi — APPROVED** [review]

File: rfcs/text/i18n.md
Comment (jamime) [reply rfcs/text/i18n.md:119]: I cant use `//` because its in the JSX, the other comments are single line so I used `//`.

File: rfcs/text/i18n.md
Comment (jamime) [reply rfcs/text/i18n.md]: Good point, I've added details about using `i18next` as a custom locale and removed references to locales other than `enGB`.

**Review by chrisbarber86 — APPROVED** [review]

**Comment by jamime** [comment]
Entering the final comment period which ends on 2021-03-26 1300.

**Review by mmaciorowski-sage — DISMISSED** [review]

**Review by philearley — DISMISSED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Comment by jamime** [comment]
Squashing commits before merge.

**Review by sjohnson-sage — APPROVED** [review]

**Pull request merged**
